projects/eval-pqmon/src/common/iio_pqm.c

```diff
@@ -514,11 +529,220 @@ int read_ch_attr(void *device, char *buf, uint32_t len,
 				       buf, len, "%" PRIu32 "",
 				       desc->pqm_ch_attr[channel->ch_num + VOLTAGE_CH_NUMBER][attr_id]);
 		}
-
+	case IIO_COUNT:
+		switch (channel->ch_num) {
+		case 0: // Dips
+			switch (attr_id) {
+			case CHAN_EVENT_COUNT:
+				return snprintf(buf, len, "%" PRIu8 "", pqlibExample.output->events.dipCount);
+			case CHAN_EVENT_START_TIME:
+				strcpy(buf, "");
+				for (int i = 0; i < MAX_EVENT_NUMBER; i++) {
+					sprintf(buffTmp, "%" PRId64 "",
+						(i < ADI_PQLIB_CFG_NUM_DIPS) ? pqlibExample.output->events.dips[i].startTime :
+						0);
+					strcat(buf, buffTmp);
+					if (i != MAX_EVENT_NUMBER - 1)
+						strcat(buf, " ");
+				}
+				return (strlen(buf));
+			case CHAN_EVENT_END_TIME:
+				strcpy(buf, "");
+				for (int i = 0; i < MAX_EVENT_NUMBER; i++) {
```

Comment:
There is separate documentation that provides details of the data process. According to it: "If there are more events detected than the buffer size, they are discarded." In this case, the previously called library function: `adi_pqlib_ProcessOneCycle ` will return ADI_PQLIB_RESULT_DIP_OUTPUT_OVERFLOW.

However, in any case, the array is a zero-initialized array for every element, containing the maximum number of possible elements. These are set to zero at every processing cycle.

Reply:
it would be good to expose the overflow through iio, i'm sure somebody will ask for it

but i'm not asking you do it now, it can be done later upon request